lib/Optimizer/Transforms/ApplyOpSpecialization.cpp

```diff
@@ -68,8 +68,14 @@ struct ApplyOpAnalysis {
       ApplyVariants variant;
       auto callee = lookupCallee(apply);
       auto iter = infoMap.find(callee);
-      if (iter != infoMap.end())
+      if (iter != infoMap.end()) {
         variant = iter->second;
+        /// NOTE: As per the `DenseMap` API  in
+        /// llvm/include/llvm/ADT/DenseMap.h, an `insert` operation doesn't
+        /// update the value if key is already in the map. Hence, remove it and
+        /// insert a new entry with updated key.
+        infoMap.erase(iter);
```

Comment:
I don't think we need to erase anything in the infoMap. Just change the insert line to use `operator[]` and update the map in place. That'll avoid freeing and allocating these structs in the map and be slightly more efficient.

Reply:
Addressed in 9e6fc7e3cf41b4d1cd529384078170664446faa7